samples/Gallery/Pages/TreeView/EditableTreeView.fs

```diff
@@ -0,0 +1,280 @@
+
+        /// Retrieves the parent Node Name from the Node Name of an Add leaf Node.
+        let getParentNodeName (node: EditableNode) = node.Name.Substring(prefix.Length)
+
+    let view () =
+        Component(program) {
+            let! model = Mvu.State
+
+            let rec filter (nodes: EditableNode list) =
+                nodes
+                |> List.filter(fun node ->
+                    node.Name.Contains(model.Filter, System.StringComparison.InvariantCultureIgnoreCase)
+                    || filter(node.Children) |> List.isEmpty |> not)
+
+            Dock() {
+                (*  TODO feeding this builder with a transient source
+                    (like you'd want to do for filtering or sorting the input nodes here)
+                    will re-render the TreeView on every interaction. Can this be avoided?
+                    Is there a more elegant (maybe even declarative) way to preserve tree node expansion?
+                    Or is there a better way to filter or sort? *)
+                TreeView(
+                    AddLeaf.appendTo (model.Nodes |> filter) None,
+                    (fun node ->
+                        if AddLeaf.is node then
+                            null
+                        else
+                            AddLeaf.appendTo (node.Children |> filter) (Some node)),
+                    (fun node ->
+                        HStack(5) {
+                            if AddLeaf.is node then
+                                Button("+", AddNodeTo(AddLeaf.getParentNodeName node))
+                                    .tip(ToolTip("Add a node"))
+                            else
+                                EditableNodeView
+                                    .view(node)
+                                    .horizontalAlignment(HorizontalAlignment.Left)
+
+                                Button("x", RemoveNode node).tip(ToolTip("Remove"))
+                        })
+                )
+                    .onSelectionChanged(SelectionItemChanged)
+                    .dock(Dock.Left)
+                    (*  TODO This solution feels awkward because it requires XAML styles,
+                        uses a TwoWay Binding against a mutable node model
+                        and the name of the EditableNode.IsExpanded property is leaking out of this module.
+                        Can either problem be avoided? *)
+                    (*  Include styles binding Avalonia.Controls.TreeViewItem.IsExpanded to EditableNode.IsExpanded
+                        to preserve tree node expansion on re-render, which is triggered by building the TreeView
+                        from a new transient collection returned by the filter method above.
+
+                        See https://github.com/AvaloniaUI/Avalonia/discussions/13903
+                        and https://github.com/AvaloniaUI/Avalonia/discussions/12397 *)
+                    .styles([ "avares://Gallery/Styles/EditableTreeView.xaml" ])
```

Comment:
```suggestion
TreeViewItem(
                            HStack(5) {
                                if AddLeaf.is node then
                                    Button("+", AddNodeTo(AddLeaf.getParentNodeName node))
                                        .tip(ToolTip("Add a node"))
                                else
                                    EditableNodeView
                                        .view(node)
                                        .horizontalAlignment(HorizontalAlignment.Left)

                                    Button("x", RemoveNode node).tip(ToolTip("Remove"))
                            }
                        ).isHitTestVisible(false)
                         .isExpanded(node.IsExpanded)
```

Reply:
This will use the newly added `TreeViewItem` widget (You will need to grab the latest changes from main). Removes in theory need for the `xaml` style.